config/config.go

```diff
@@ -1896,6 +1935,36 @@ func (c *Config) getFieldTagFilter(tbl *ast.Table, fieldName string) []models.Ta
 	return target
 }
 
+func (*Config) getFieldMap(tbl *ast.Table, fieldName string) map[string]string {
+	target := make(map[string]string)
+	if node, ok := tbl.Fields[fieldName]; ok {
+		if subTbl, ok := node.(*ast.Table); ok {
+			for _, val := range subTbl.Fields {
+				if kv, ok := val.(*ast.KeyValue); ok {
+					if str, ok := kv.Value.(*ast.String); ok {
+						target[kv.Key] = str.Value
+					}
+				}
+			}
+		}
+	}
+
+	return target
+}
+
+func (c *Config) matchesLabelSelection(pluginType, name string, tbl *ast.Table) (bool, error) {
+	// Get the label definitions for the plugin and check them
+	labels := c.getFieldMap(tbl, "labels")
+	for k, v := range labels {
+		if err := CheckSelectionKeyValuePairs(k, v); err != nil {
+			return false, fmt.Errorf("invalid label in plugin %s.%s: %s=%q: %w", pluginType, name, k, v, err)
```

Comment:
If you construct the error message in the caller side we don't need to pass all those parameters!

Reply:
That is a good point. However, since the message would most likely be same except the plugin type, we can avoid duplication if it comes from this func.

Let me know your thoughts.